### ISSUE TITLE
Doesn't go on top of keyboard

### ISSUE THREAD

**Issue by pjebs** [nimati/FCAlertView#11]
In my use case, the FCAlertView does not go on top of the keyboard.
I have a similar component that shows a progress circle that is used in exactly the same circumstance, but that goes on top of the keyboard. This is causing issues because whilst the AlertView is operating, the user can still type stuff in keyboard.

I looked at code:

Your code:

```
- (void) showAlertInView:(UIViewController *)view withTitle:(NSString *)title withSubtitle:(NSString *)subTitle withCustomImage:(UIImage *)image withDoneButtonTitle:(NSString *)done andButtons:(NSArray *)buttons {

    [self checkCustomizationValid];
    [self safetyCloseCheck];

    [view.view.window addSubview:self];

}
```

I looked at other component's code: https://github.com/AssistoLab/KVNProgress/blob/master/KVNProgress/Classes/KVNProgress.m#L870

```
- (void)addToWindow
{
    self.originalKeyWindow = [UIApplication sharedApplication].keyWindow;

    if (!self.progressWindow) {
        self.progressWindow = [[UIWindow alloc] initWithFrame:self.originalKeyWindow.frame];

        // That code makes the custom UIWindow handle the orientation changes.
        // http://stackoverflow.com/a/27091111/2571566
        self.progressWindow.rootViewController = [[KVNRotationViewController alloc] init];
    }

    self.progressWindow.frame = self.originalKeyWindow.frame;

    // Since iOS 9.0 set the windowsLevel to UIWindowLevelStatusBar is not working anymore.
    // This trick, place the progressWindow on the top.
    UIWindow *lastWindow = [[[UIApplication sharedApplication] windows] lastObject];
    self.progressWindow.windowLevel = lastWindow.windowLevel + 1;

    [self.progressWindow makeKeyAndVisible];
    [self addToView:self.progressWindow];
}
```

`KVNProgress` gives us the option to select the view we want the component to be on top of. If we don't select a view, it uses the code above to correctly place the component on top of everything - including the keyboard.

**This is the approach to take.** Can you please make another method that allows _us_ to provide the view instead of us providing the viewcontroller.

**Comment by nimati** [comment]
Haa, interesting. Good catch! I'll look into it and add a fix soon for this, @pjebs.

**Comment by pjebs** [comment]
any news on this?

**Comment by nimati** [comment]
Hey, not yet, will update you when it's ready.

**Comment by pjebs** [comment]
https://github.com/AssistoLab/KVNProgress/issues/70

**Comment by nimati** [comment]
@pjebs I'll be taking a look in the coming days and have a new release to fix it.

**Comment by nimati** [comment]
@pjebs Just released FCAlertView V1.0.8 which should now fix this issue. To present the alert on top of your keyboard, use:

``` Objective-C
[alert showAlertWithTitle:@"Alert Title"
          withSubtitle:@"This is your alert's subtitle. Keep it short and concise. 😜👌"
       withCustomImage:nil
   withDoneButtonTitle:nil
            andButtons:nil];
```

This method will make the alert appear on top of all other UI elements on the main window of the UIApplication. Let me know if it works for you.

Cheers!

**Comment by nimati** [comment]
any updates @pjebs ?

**Comment by pjebs** [comment]
I'll check it out soon.

**Comment by pjebs** [comment]
Didn't work.

`showAlertInView` - Fine
`showAlertInWindow` - Fine
`showAlertWithTitle` -Not fine

This attempts to add to the current keyWindow.
If the keyboard is up, it won't work.

What this method should do is **create a new UIWindow**, set it's windowLevel using the trick KVNProgress does. Make this newly created UIWindow as as the keyWindow etc.

If we want the currently implemented functionality, we can just use `showAlertInWindow` easily.

**Comment by nimati** [comment]
So you're saying only showAlertWithTitle creates the problem but the other methods for showing works?

**Comment by pjebs** [comment]
yes

**Comment by pjebs** [comment]
`showAlertWithTitle` works fine too except when there is a keyboard.

**Comment by nimati** [comment]
Are you sure? I did test with `showAlertWithTitle` with a keyboard displayed and it worked just fine: 

![simulator screen shot sep 18 2016 2 21 10 am](https://cloud.githubusercontent.com/assets/7935593/18613347/9bd4672c-7d46-11e6-95dd-5916451eb238.png)

**Comment by pjebs** [comment]
That;s interesting. In mine, the keyboard is on top. I'm on iOS9.3

**Comment by pjebs** [comment]
I assume it must be related to: https://github.com/AssistoLab/KVNProgress/issues/86

In which case your component is fine!

**Comment by nimati** [comment]
Ha, I see. Well keep me posted if any other issues come up. Cheers! 👍

**Comment by pjebs** [comment]
There are some oddities with this.

The source of the issue I raised is related to how the component places itself at the top.
I found that when I stopped using KVNProgress, it worked. Then when I used [CNPPopupController](https://github.com/carsonperrotti/CNPPopupController) It again was behind the keyboard.

Both KVNProgress and CNPPopupController become topmost.

My solution is I can't directly use `showAlertWithTitle` - which works when the keyboard is above.

My workaround is:
- If the keyboard is up: Use `showAlertWithTitle` (when keyboard is **not** up, it doesn't even show when CNPPopupController is on top. Presumably the alertbox is actually behind  CNPPopupController)
- If keyboard is not up: Use `showAlertInView`.
